mozetl/taar/taar_amowhitelist.py

```diff
@@ -16,18 +18,112 @@
 AMO_DUMP_BUCKET = 'telemetry-parquet'
 AMO_DUMP_PREFIX = 'telemetry-ml/addon_recommender/'
 
+# Input file
 AMO_DUMP_BASE_FILENAME = 'extended_addons_database'
+AMO_DUMP_FILENAME = AMO_DUMP_BASE_FILENAME + '.json'
+
+# Output files
 FILTERED_AMO_BASE_FILENAME = 'whitelist_addons_database'
+FEATURED_BASE_FILENAME = 'featured_addons_database'
+FEATURED_WHITELIST_BASE_FILENAME = 'featured_whitelist_addons'
 
-AMO_DUMP_FILENAME = AMO_DUMP_BASE_FILENAME + '.json'
 FILTERED_AMO_FILENAME = FILTERED_AMO_BASE_FILENAME + '.json'
+FEATURED_FILENAME = FEATURED_BASE_FILENAME + '.json'
+FEATURED_WHITELIST_FILENAME = FEATURED_WHITELIST_BASE_FILENAME + '.json'
 
 MIN_RATING = 3.0
 MIN_AGE = 60
 
 logger = logging.getLogger('amo_whitelist')
 
 
+class AbstractAccumulator:
+    def __init__(self):
+        self._results = {}
+
+    @abstractmethod
+    def process_record(self, guid, addon_data):
+        pass
+
+    def get_results(self):
+        return self._results
+
+
+class FeaturedAccumulator(AbstractAccumulator):
+    def __init__(self):
+        AbstractAccumulator.__init__(self)
+
+    def process_record(self, guid, addon_data):
+
+        featured = addon_data.get('is_featured', False)
+
+        if featured:
+            self._results[guid] = addon_data
+
+
+class WhitelistAccumulator(AbstractAccumulator):
+    def __init__(self, min_age, min_rating):
+        AbstractAccumulator.__init__(self)
+
+        self._min_age = min_age
+        self._min_rating = min_rating
+
+        self._latest_create_date = datetime.datetime.today() - \
+                                   datetime.timedelta(days=self._min_age)  # noqa
+        self._latest_create_date = self._latest_create_date.replace(tzinfo=None)
+
+    def process_record(self, guid, addon_data):
+        if guid == 'pioneer-opt-in@mozilla.org':
+            # Firefox Pioneer is explicitly excluded
+            return
+
+        current_version_files = addon_data.get('current_version', {}).get('files', [])
+        if len(current_version_files) == 0:
+            # Only allow webextensions
+            return
+
+        if current_version_files[0].get('is_webextension', False) is False:
+            # Only allow webextensions
+            return
+
+        rating = addon_data.get('ratings', {}).get('average', 0)
+        create_date = parse(addon_data.get('first_create_date', None)).replace(tzinfo=None)
+
+        if rating >= self._min_rating and create_date <= self._latest_create_date:
+            self._results[guid] = addon_data
+
+
+class WhitelistFeaturedAccumulator(WhitelistAccumulator):
+    def __init__(self, min_age, min_rating):
+        WhitelistAccumulator.__init__(self, min_age, min_rating)
+
+    def process_record(self, guid, addon_data):
```

Comment:
Could this be simplified?

```python
    def process_record(self, guid, addon_data):
        if not addon_data.get('is_feature'):
            return
        return super(WhitelistFeatureAccumulator, self).process_record(guid, addon_data)
```

Reply:
Whooops.  That's embarassing.  I've changed this now, but I invoked the method without super.  I'm just invoking the class method directly and passing in self so that I don't get Python3 breakage whenever that comes down the line.